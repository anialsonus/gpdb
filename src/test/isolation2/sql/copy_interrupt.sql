!\retcode gpconfig -c client_connection_check_interval -v 20s;
!\retcode gpstop -u;

0: CREATE TABLE copy_interrupt_table(a int);
0&: COPY copy_interrupt_table FROM PROGRAM 'while true; do echo 1; sleep 1; done | cat -';
! sleep 10;

0t:
! sleep 40;

SELECT pg_terminate_backend(pid)
    FROM pg_stat_activity where query LIKE 'COPY copy_interrupt_table FROM PROGRAM%';

DROP TABLE copy_interrupt_table;
!\retcode gpconfig -c client_connection_check_interval -v 0;