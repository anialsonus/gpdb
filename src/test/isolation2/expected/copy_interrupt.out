!\retcode gpconfig -c client_connection_check_interval -v 20s;
-- start_ignore

-- end_ignore
(exited with code 0)
!\retcode gpstop -u;
-- start_ignore

-- end_ignore
(exited with code 0)

0: CREATE TABLE copy_interrupt_table(a int);
CREATE
0&: COPY copy_interrupt_table FROM PROGRAM 'while true; do echo 1; sleep 1; done | cat -';  <waiting ...>
! sleep 10;


0t: ... <terminating>
! sleep 40;


SELECT pg_terminate_backend(pid) FROM pg_stat_activity where query LIKE 'COPY copy_interrupt_table FROM PROGRAM%';
 pg_terminate_backend 
----------------------
(0 rows)

DROP TABLE copy_interrupt_table;
DROP
!\retcode gpconfig -r client_connection_check_interval;
-- start_ignore

-- end_ignore
(exited with code 0)
